sources/shopify_dlt/helpers.py

```diff
@@ -1,23 +1,93 @@
 """Shopify source helpers"""
 
-from dlt.common.typing import TDataItem
-from typing import Any, Iterable
+from dlt.sources.helpers import requests
+from dlt.common.typing import TDataItem, TDataItems, Dict
+from typing import Any, Iterable, Optional, Literal
 
+from .settings import DEFAULT_API_VERSION
+from .date_helper import ensure_pendulum_datetime
 
-def iterate_page(page: Any) -> Iterable[TDataItem]:
+TOrderStatus = Literal["open", "closed", "cancelled", "any"]
+
+
+class ShopifyApi:
+    """
+    A Shopify API client that can be used to get pages of data from Shopify.
     """
-    Iterates over all pages yielding all items
 
-    Args:
-        page (Any): The starting page object.
+    def __init__(
+        self,
+        shop_url: str,
+        private_app_password: str,
+        api_version: str = DEFAULT_API_VERSION,
+    ) -> None:
+        """
+        Args:
+            shop_url: The URL of your shop (e.g. https://my-shop.myshopify.com).
+            private_app_password: The private app password to the app on your shop.
+            api_version: The API version to use (e.g. 2023-01)
+        """
+        self.shop_url = shop_url
+        self.private_app_password = private_app_password
+        self.api_version = api_version
 
-    Yields:
-        TDataItem: Each item from the pages.
+    def get_pages(
+        self, resource: str, params: Optional[Dict[str, Any]] = None
+    ) -> Iterable[TDataItems]:
+        """Get all pages from shopify using requests.
+        Iterates through all pages and yield each page items.
 
-    Returns:
-        Iterable[TDataItem]: An iterable of items from the pages.
-    """
-    while page:
-        for item in page:
-            yield item.to_dict()
-        page = page.next_page() if page.has_next_page() else None
+        Args:
+            resource: The resource to get pages for (e.g. products, orders, customers).
+            params: Query params to include in the request.
+
+        Yields:
+            List of data items from the page
+        """
+        url = f"{self.shop_url}/admin/api/{self.api_version}/{resource}.json"
+        headers = {"X-Shopify-Access-Token": self.private_app_password}
+        while url:
+            response = requests.get(url, params=params, headers=headers)
+            response.raise_for_status()
+            json = response.json()
+            # Get item list from the page
+            yield [self._convert_datetime_fields(item) for item in json[resource]]
+            link = response.headers.get("Link")
```

Comment:
`requests` have links parsing built in: https://requests.readthedocs.io/en/latest/user/advanced/#link-headers
also see github: _get_rest_pages

Reply:
Aha, that it does!